shell/platform/windows/dpi_utils.h

```diff
@@ -0,0 +1,23 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "Windows.h"
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_DPI_UTILS_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_DPI_UTILS_H_
+
+namespace flutter {
+
+/// Returns the current DPI. Supports all DPI awareness modes, and is backward
+/// compatible down to Windows Vista. If |hwnd| is nullptr, returns the DPI for
+/// the primary monitor. Otherwise, returns the system's DPI.
+UINT GetDpiForHWND(HWND hwnd);
+
+/// Returns the DPI of a given monitor. Defaults to 96 if the API is not
+/// available.
+UINT GetDpiForMonitor(HMONITOR monitor);
+
```

Comment:
I know it's tested indirectly already, but there should be a `dpi_utils_unittests.cc` testing this directly. It can do the simple tests that both return non-zero values, and also validate the nullptr handling above by ensuring that `GetDpiForHWND(nullptr)` returns the same things as `GetDpiForMonitor` with the primary monitor.